### ISSUE TITLE
[BUG] [JAVA] content type in native client is url-escaped, charset=utf-8 doesn't work properly

### ISSUE THREAD

**Issue by starwarswii** [OpenAPITools/openapi-generator#17864]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

when specifiying a content with an equals sign, e.g.:
```yaml
content:
  application/json;charset=utf-8:
    schema: ...
```

the content type seems to be url-encoded, leading to code like this in the generated client:
```java
localVarRequestBuilder.header("Content-Type", "application/json;charset&#x3D;utf-8");
localVarRequestBuilder.header("Accept", "application/json;charset&#x3D;utf-8");
```

note the `=` becoming `&#x3D;`

this is incorrect, and leads to servers not handling the request properly

##### openapi-generator version

tested on:
```
openapi-generator-cli-7.3.0.jar
openapi-generator-cli-7.4.0-20240214.132808-11.jar
openapi-generator-cli-8.0.0-20231011.125706-1.jar
```

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.1
info:
  title: OpenAPI Petstore
  description: sample
  version: 1.0.0
tags: []
paths:
  /request:
    get:
      operationId: someOperation
      tags:
      - sometag
      requestBody:
        description: body
        content:
          application/json;charset=utf-8:
            schema:
              type: string
      responses:
        '200':
          description: response
          content:
            application/json;charset=utf-8:
              schema:
                type: string
```

##### Generation Details

ran generator via
```
java -jar openapi-generator-cli-<version>.jar generate -g java -i test_yaml.yaml -p library=native
```

##### Steps to reproduce

1. download newest `openapi-generator-cli` build
2. create `test_yaml.yaml` with above content
3. run generator as shown above
4. inspect `src/main/java/org/openapitools/client/api/SometagApi.java`.<br>at the bottom, inside the method `someOperationRequestBuilder`, you will see these lines:
    ```java
    localVarRequestBuilder.header("Content-Type", "application/json;charset&#x3D;utf-8");
    localVarRequestBuilder.header("Accept", "application/json;charset&#x3D;utf-8");
    ```

##### Suggest a fix

the expected output should be
```java
localVarRequestBuilder.header("Content-Type", "application/json;charset=utf-8");
localVarRequestBuilder.header("Accept", "application/json;charset=utf-8");
```

a fix is to not urlencode the content-type and accept fields. at the least not encoding equals signs

**Comment by wing328** [comment]
thanks for reporting the issue. I've filed 17872 to fix it.

**Comment by wing328** [comment]
fix merged. please give it a try with the latest master or snapshot version

**Comment by starwarswii** [comment]
thanks! confirmed it's fixed on `openapi-generator-cli-7.4.0-20240216.074004-22.jar`
